digestwriter/consumer.go

```diff
@@ -244,8 +252,45 @@ func extractDigestsFromMessage(workload Workload) (digests []string) {
 	return
 }
 
+// isMessageInGzipFormat will check if the format of the message is gzip if it is it will return true if not it will return false
+func isMessageInGzipFormat(messageValue []byte) bool {
+	if messageValue == nil {
+		return false
+	}
+	if len(messageValue) < 2 {
+		return false
+	}
+	// Checking for first 2 bytes in gzip instance witch are 31 and 139
+	if messageValue[0] == gzip1 && messageValue[1] == gzip2 {
+		return true
+	}
+	return false
+
+}
+
+// decompressMessage will try to decompress the message if the message is compressed
+func decompressMessage(messageValue []byte) []byte {
+	if isMessageInGzipFormat(messageValue) {
+		reader := bytes.NewReader(messageValue)
+		gzipReader, err := gzip.NewReader(reader)
+		if err != nil {
+			logger.Errorf("failed to read the compressed message: %s", err.Error())
+			return nil
+		}
+		defer gzipReader.Close()
+		decompresed, err := ioutil.ReadAll(gzipReader)
+		if err != nil {
+			logger.Errorf("failed to decompress the message: %s", err.Error())
+			return nil
+		}
+		return decompresed
+	}
+	return messageValue
+}
```

Comment:
I would do it this way and log error in caller function.
```suggestion
func decompressMessage(messageValue []byte) ([]byte, error) {
	if isMessageInGzipFormat(messageValue) {
		reader := bytes.NewReader(messageValue)
		gzipReader, err := gzip.NewReader(reader)
		if err != nil {
			return nil, err
		}
		defer gzipReader.Close()
		decompresed, err := ioutil.ReadAll(gzipReader)
		if err != nil {
			return nil, err
		}
		return decompresed
	}
	return messageValue
}
```
Please use `ammend` or `fixups` locally to apply changes.